src/Autodesk.Forge.DesignAutomation/Model/ArgumentConverter.cs

```diff
@@ -33,25 +33,34 @@ public override IArgument Create(Type objectType)
 
         public override object ReadJson(JsonReader reader, Type objectType, object existingValue, JsonSerializer serializer)
         {
+            IArgument target;
             if (reader.TokenType == JsonToken.StartObject)
             {
-                IArgument target;
-                var jObject = JObject.Load(reader);
-                if (jObject.Property("url") != null)
+                JObject jObject = JObject.Load(reader);
+                if (jObject["url"] != null)
                 {
                     target = new XrefTreeArgument();
+                    serializer.Populate(jObject.CreateReader(), target);
+                }
+                else if (jObject["value"] != null)
+                {
+                    target = new StringArgument(jObject["value"].Value<string>());
                 }
                 else
                 {
-                    target = new StringArgument();
+                    throw new JsonSerializationException("Expected XrefTreeArgument or StringArgument.");
```

Comment:
Why are we throwing a different exception here than below? I'd prefer if we threw the same type of exception for all  parsing errors

Reply:
I think there are different exception type, this one means json type is accepted, but they cannot be serialized as  XrefTreeArgument or StringArgument

Reply:
The other exception means even the json type is wrong, we don't accept them at all